src/Network/Network/VirtualNetwork/Subnet/AzureVirtualNetworkSubnetConfigBase.cs

```diff
@@ -61,18 +62,22 @@ public class AzureVirtualNetworkSubnetConfigBase : NetworkBaseCmdlet
             HelpMessage = "RouteTable")]
         public PSRouteTable RouteTable { get; set; }
 
+        [GenericBreakingChange("Update Property Name", OldWay = "-ResourceId", NewWay = "-NatGatewayId")]
+        [Alias("NatGatewayId")]
         [Parameter(
             Mandatory = false,
             ValueFromPipelineByPropertyName = true,
             ParameterSetName = "SetByResourceId",
-            HelpMessage = "ResourceId")]
+            HelpMessage = "Specifies the Id of NAT Gateway resource associated with the subnet configuration")]
         public string ResourceId { get; set; }
 
+        [GenericBreakingChange("Update Property Name", OldWay = "-InputObject", NewWay = "-NatGateway")]
```

Comment:
We shouldn't have a breaking change attribute as we won't remove InputObject since it is a standard parameter for Azure PowerShell cmdlets.

Reply:
this was agreed upon in offline meeting with @niander  and @markcowl. We will retain inputobject on natgateway operations but can change parameter name when it is referenced from another operation. (Similar to RouteTable here) @MiYanni

Reply:
Discussed with Niander. Looks good.